### PR TITLE
Slack filter to ignore any empty errorreports

### PR DESCRIPTION
**Pull request by DanielMurphy22**
Only if an errorreport has a non-default entry (such as Not provided) for the stacktrace, the textbox or the email will a notiication be sent to slack. With none of these we cannot resolve the problem.

To test:

On a local server (with mantid.properties: `errorreports.rooturl = http://localhost:8082`) and the .env file has the Slack line un-commented. Make sure to use Mantid workbench for producing errorreporter as only it can send a non-empty stacktrace.
check:
- No notification sent to Slack if Email, text and stacktrace are all "Not provided". Test this by loading `TrainingCourseData/Training_Exercise3a_SNS`, executing `NormaliseToMonitor` and clicking "Share Non-identifiable Info"
- Yes a notification is sent to Slack if Email and text are "Not provided", but Stacktrace is full. Test this by loading `TrainingCourseData/Training_Exercise3a_SNS`, executing `NormaliseToMonitor` and clicking "Yes, Share Info"
- Yes a notification is sent to Slack if Email and Stacktrace are "Not provided", but text is full. Test this by executing `Segfault`, setting text = Test and clicking "Yes, Share Info"
- Yes a notification is sent to Slack if Text and Stacktrace are "Not provided", but email is full. Test this by executing `Segfault`, setting email = Test and clicking "Yes, Share Info"

### CODE REVIEW

File: web/services/migrations/0006_slack_filter.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.2.11 on 2020-03-19 12:34
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```
Comment (martyngigg) [inline web/services/migrations/0006_slack_filter.py]: I assume I am misunderstanding something. The model has no changes to the fields so I wasn't expecting a migration?

File: web/services/models.py [hunk 2]
```diff
@@ -65,14 +65,16 @@ def notify_report_received(sender, instance, signal, *args, **kwargs):
...
     """
-    if instance.user is None:
-        return
 
     email = instance.user.email
-    if email == TEST_EMAIL:
-        # Don't send a notification if there was not email provided as we can't
-        # actively do anything about it
-        return
+    if email in TEST_EMAIL:
```
Comment (martyngigg) [inline web/services/models.py]: The nested ifs are little hard to read. Could we write this as
```python
# Don't send a slack notification if there was not 
# a useful email, stacktrace or text provided 
# as we can't actively do anything about it
for field in (email, instance.textBox, instance.stacktrace):
	if field in TEST_VALUES:
		return
```
and change `TEST_EMAIL` to `TEST_VALUES` as they are no longer single thing?

**Comment by DanielMurphy22** [comment]
> Thanks for looking at this. Was it the nightly system tests producing that spam?

Yes it was. This should stop them being posted to Slack but is it worth stopping them from being sent to the error reporter server in the first place or does that defeat their purpose?

I didn't expect a migration file but decided to check for one and found this. I have removed it as I don't think it makes sense.

**Comment by martyngigg** [comment]
> > Thanks for looking at this. Was it the nightly system tests producing that spam?
> 
> Yes it was. This should stop them being posted to Slack but is it worth stopping them from being sent to the error reporter server in the first place or does that defeat their purpose?

We wanted to make sure it was tested so we would know if it fell over. I wonder if we need this now we have the errors going to Slack. @Mantid-Matthew  What do you think?

> I didn't expect a migration file but decided to check for one and found this. I have removed it as I don't think it makes sense.

@Mantid-Matthew Could this migration actually be from the other alterations?

File: web/services/models.py (hunk 2 above)
Comment (DanielMurphy22) [reply web/services/models.py]: I added this suggestion to my second commit but I'm not sure this is what I want. The nested ifs only blocked a slack notification if all 3 fields were in TEST_VALUES. The for-if suggestion blocks a slack notification if any one of the 3 fields were in TEST_VALUES, which is not my intention.

**Comment by Mantid-Matthew** [comment]
I think I actually forgot to commit the migration file from the previous set of alterations so could well be. The migration files aren't actually needed as Django will auto-generate them for you but might be worth including.

**Comment by Mantid-Matthew** [comment]
I think now that they are going to slack we can probably stop the systemtest. It was always a bit odd as it could stop the nightly builds going through if the server was down anyway.

**Comment by martyngigg** [comment]
> I think now that they are going to slack we can probably stop the systemtest. It was always a bit odd as it could stop the nightly builds going through if the server was down anyway.

Agreed. I'll make an issue.

**Comment by martyngigg** [comment]
Issue for system test removal: mantidproject/mantid#28397

File: web/services/models.py [hunk 3]
```diff
@@ -65,14 +65,18 @@ def notify_report_received(sender, instance, signal, *args, **kwargs):
     :param args: Unused
     :param kwargs: Unused
     """
-    if instance.user is None:
```
Comment (martyngigg) [inline web/services/models.py]: @DanielMurphy22 I think this change has called the test failure. When `instance.user` is `None` then you'll get a problem with doing `instance.user.email` later.

File: web/services/models.py [hunk 4]
```diff
@@ -65,7 +65,9 @@ def notify_report_received(sender, instance, signal, *args, **kwargs):
     :param args: Unused
     :param kwargs: Unused
     """
-
+    if instance.user is None:
```
Comment (martyngigg) [inline web/services/models.py]: Did you want the stack trace in this case?

Comment (DanielMurphy22) [reply web/services/models.py]: Yes

Comment (DanielMurphy22) [reply web/services/models.py]: will I need this none protection for instance.stacktrace and instance.textBox?

**Comment by martyngigg** [comment]
I will get this deployed now.

**Pull request merged**